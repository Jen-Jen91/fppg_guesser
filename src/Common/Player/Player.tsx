import React from "react";
import { roundOffPoints } from "../helpers";
import "./Player.scss";

export interface PlayerProps {
  playerFirstName: string;
  playerLastName: string;
  url: string;
  fppg: number;
  showFppg: boolean;
  clickPlayer: (fppg: number) => void;
}

const Player = (props: PlayerProps) => {
  const {
    playerFirstName,
    playerLastName,
    url,
    fppg,
    showFppg,
    clickPlayer
  } = props;

  return (
    <div onClick={() => clickPlayer(fppg)} className="player-container">
      <img
        src={url}
        alt={playerFirstName + " " + playerLastName}
        width={300}
        height={300}
      />

      <p className="name-text">
        {playerFirstName} {playerLastName}
      </p>

      {showFppg && <p className="fppg-text">Fppg: {roundOffPoints(fppg)}</p>}
    </div>
  );
};

export default Player;
