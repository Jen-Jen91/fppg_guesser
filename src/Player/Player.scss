@import "../Common/variables.scss";

.player-container {
  @include flex-column(center, flex-start);

  border: solid 5px $light-blue;
  border-radius: 10px;

  cursor: pointer;

  &:hover {
    background-color: $light-blue;
  }
}

.name {
  margin: 15px 0;

  font-size: 1.5rem;
  font-weight: bold;
  color: $dark-blue;
}

.fppg-text {
  @extend .name;

  margin: 0 0 10px 0;

  font-size: 0.9rem;
  text-transform: uppercase;
}
