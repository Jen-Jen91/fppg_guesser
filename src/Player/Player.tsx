import React from "react";

export function roundOffPoints(points: number) {
  return points.toFixed(2);
}

export interface PlayerProps {
  playerFirstName: string;
  playerLastName: string;
  url: string;
  fppg: number;
  clickPlayer: (fppg: number) => void;
  clickDisabled: boolean;
  showFppg: boolean;
}

export default function Player(props: PlayerProps) {
  const {
    playerFirstName,
    playerLastName,
    url,
    fppg,
    clickPlayer,
    clickDisabled,
    showFppg
  } = props;

  console.log("CLICK IS DISABLED?: ", clickDisabled);

  return (
    <div onClick={() => clickPlayer(fppg)} className="player-container">
      <img src={url} alt={playerFirstName + " " + playerLastName} />
      <p>
        {playerFirstName} {playerLastName}
      </p>
      {showFppg && <p className="fppg-text">{roundOffPoints(fppg)}</p>}
    </div>
  );
}
