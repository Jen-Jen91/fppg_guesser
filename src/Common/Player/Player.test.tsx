import React from "react";
import Player from "./Player";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

let playerClicked = jest.fn();

describe("Player", () => {
  test("snapshot test", () => {
    const tree = renderer
      .create(
        <Player
          playerFirstName="John"
          playerLastName="Doe"
          url=""
          fppg={48}
          clickPlayer={playerClicked}
          showFppg={false}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders fppg text when true", () => {
    const wrapper = mount(
      <Player
        playerFirstName="John"
        playerLastName="Doe"
        url=""
        fppg={48}
        clickPlayer={playerClicked}
        showFppg={true}
      />
    );

    expect(wrapper.find(".fppg-text")).toHaveLength(1);
  });

  test("calls clickPlayer when clicked", () => {
    const wrapper = mount(
      <Player
        playerFirstName="John"
        playerLastName="Doe"
        url=""
        fppg={48}
        clickPlayer={playerClicked}
        showFppg={false}
      />
    );

    const div = wrapper.find(".player-container");
    div.simulate("click");

    expect(playerClicked).toBeCalledWith(48);
  });
});
