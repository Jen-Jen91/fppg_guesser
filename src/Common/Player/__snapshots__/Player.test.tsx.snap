// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player snapshot test 1`] = `
<div
  className="player-container"
  onClick={[Function]}
>
  <img
    alt="John Doe"
    height={300}
    src=""
    width={300}
  />
  <p
    className="name-text"
  >
    John
     
    Doe
  </p>
</div>
`;
